### PR TITLE
feat: migrate payment enablement from toggle to delete button

### PR DESCRIPTION
**Pull request by foochifa**
## Problem

We want to remove the toggle button, and instead follow the usual deletion method of our form fields. 

Closes FRM-1016

## Solution

I considered two potential options

1. Remove `payments_field.enabled` and couple the 'enablement' of the field with the existence of `payments_field`. This will require changes to the `form` schema, and we will have to create a new ` DELETE .../payments` endpoint. However, this method will align closest to the idea of field creation / deletion.
2. Rely on `payments_field.enabled` for creating and deleting a field. Currently, by default, storage mode forms will be created with `payments_field.enabled = false`. We can capitalise on this and hard-set 'save field' to mutate `enabled = true` and for deletion we could mutate `enabled = false`

For speedy changes to the UI/UX, have decided to capitalise on option 2, which require minimal changes on the BE, and is generally a FE fix. 

Other considerations / changes:
1. Created a separate modal for deleting payments, as the delete form modal, albeit similar, is highly coupled with field builder state
2. Utilise useDeleteFormField mutation file for mutation of deleting payment fields
3. Loosen the requirement of `payment_type` in `PUT .../payments` endpoint to enable option 2 above
4. Copied a custom version of `FieldButtonGroup` from `FieldRowContainer` to `PaymentView` instead of abstracting it, due the 'duplicate' option
5. Disabled form controls in `PaymentInputPanel`, as its closer to our design, but still doesn't completely align with it (as there are slight differences in disablement between email form vs unconnected stripe storage form)
6. Due to bugs from tabbing out of payment input panel in `PaymentButtonGroup` will only display buttons if the field is created. 

**Breaking Changes** 

- [ ] No - this PR is backwards compatible. Though on rolling update, user might have a 'validation failed' toast when trying to delete payment fields. As I loosen the validation of the update payment endpoint in this PR.

## Before & After Screenshots

**AFTER**:

image
image
image

## Tests

**In desktop**
- [ ] Go to a storage mode form (stripe unconnected)
- [ ] Go to Create Tab -> Payments, you should see the preview but cannot save field
- [ ] Go to settings -> connect stripe
- [ ] Go to Create Tab -> Payments
- [ ] Create a fixed payment
- [ ] Open the form and make a payment
- [ ] Delete the payment field
- [ ] Open the form and submit the form (no payment)
- [ ] Create a variable payment
- [ ] Open the form and make a payment
- [ ] Delete the payment field
- [ ] Open the form and submit the form (no payment)

**In mobile**
- [ ] Go to a storage mode form (stripe unconnected)
- [ ] Go to Create Tab -> Payments, you should not be able to save field
- [ ] Go to settings -> connect stripe
- [ ] Go to Create Tab -> payments
- [ ] Create a fixed payment
- [ ] Edit the payment field by selecting the `gear` icon on the field
- [ ] Open the form and make a payment
- [ ] Delete the payment field
- [ ] Open the form and submit the form (no payment)

### CODE REVIEW

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/PaymentView.tsx [hunk 1]
```diff
@@ -75,36 +84,93 @@ export const PaymentView = () => {
...
-          >
-            <Box p={{ base: '0.75rem', md: '1.5rem' }}>
-              <PaymentPreview
-                colorTheme={form?.startPage.colorTheme}
-                paymentDetails={paymentDetails}
-                isBuilder
-              />
+  return (
+    (paymentDetails.enabled ||
+      paymentState === PaymentState.EditingPayment) && (
```
Comment (KenLSM) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/PaymentView.tsx]: Would prefer to use guard clauses here. This should help to reduce the indents and cognitive load as we can exclude unhappy cases earlier.
```
if (!paymentDetails.enabled && paymentState !== PaymentState.EditingPayment) { 
    return null
}

return (<Box>{...}</Box>)
```

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx [hunk 2]
```diff
@@ -156,117 +152,103 @@ const PaymentInput = ({ isDisabled }: { isDisabled: boolean }) => {
...
+      >
+        <FormLabel>Payment type</FormLabel>
+        <Controller
+          name={'payment_type'}
+          control={control}
+          render={({ field }) => (
+            <SingleSelect
+              isClearable={false}
+              placeholder="Select Payment Type"
+              fullWidth={true}
```
Comment (KenLSM) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx]: /nit not related to your changes, but could you help me to simply `fullWidth={true}` to `fullWidth`?

File: frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts [hunk 3]
```diff
@@ -85,6 +106,34 @@ export const useDeleteFormField = () => {
     [toast],
   )
 
+  const deletePaymentFieldMutation = useMutation(
+    () => updateFormPayments(formId, { enabled: false } as PaymentsUpdateDto),
```
Comment (KenLSM) [inline frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts]: This is a little yucky due to the types.

I'm thinking if it is ok to set our default as `{ enabled: false, paymentType: PaymentType.Fixed, amount_cents: 0 }` which also reflects the DB defaults -- couldn't really think of anything better 😢

Comment (KenLSM) [reply frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts]: On a related note of removing forced type casting, could I also get your help to remove `as FormPaymentsField` in [frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx:L126](https://github.com/opengovsg/FormSG/blob/feat/migrate-payment-enablement/frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx#L126C12-L126C29)?

**Review by KenLSM — APPROVED** [review]
LGTM! Some comments on typecasting.

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/PaymentView.tsx (hunk 1 above)
Comment (foochifa) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/PaymentView.tsx]: [f5f41ce](https://github.com/opengovsg/FormSG/pull/6511/commits/f5f41cea39fa3d7eeb0354169637fd99dc1d551b)!

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx (hunk 2 above)
Comment (foochifa) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx]: Thanks Ken! [cfb6493](https://github.com/opengovsg/FormSG/pull/6511/commits/cfb64937fe2bbc615ff20a389dfe687a21e43b60)

File: frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts (hunk 3 above)
Comment (foochifa) [reply frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts]: Its still a little yucky as I had to hard-type `payment_type` else typescript will complain 😩 [62baf5c](https://github.com/opengovsg/FormSG/pull/6511/commits/62baf5c825d6b37bab5a3dc513e2205bb6cdb379)

Comment (foochifa) [reply frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts]: [85f399d](https://github.com/opengovsg/FormSG/pull/6511/commits/85f399d63aefb6752741e535943f1157b57215c2) Paired thanks Ken!

**Pull request merged**